samples/encode-decode-worker/js/main.js

```diff
@@ -0,0 +1,290 @@
+  // Create a new worker.
+  streamWorker = new Worker("js/stream_worker.js");
+  addToEventLog('Worker created.');
+  // Print messages from the worker in the text area.
+  streamWorker.addEventListener('message', function(e) {
+    addToEventLog('Worker msg: ' + e.data.text, e.data.severity);
+  }, false);
+
+  stopButton.onclick = () => {
+    addToEventLog('Stop button clicked.');
+    stop();
+  }
+
+  connectButton.onclick = () => {
+    connectButton.disabled = true;
+    stopButton.disabled = false;
+    hwButtons.style.display = "none";
+    prefButtons.style.display = "none";
+    codecButtons.style.display = "none";
+    resButtons.style.display = "none";
+    modeButtons.style.display = "none";
+    rateInput.style.display = "none";
+    keyInput.style.display = "none";
+    startMedia();
+  }
+
+  async function startMedia() {
+    if (stopped) return;
+    addToEventLog('startMedia called'); 
+    try {
+      // Collect the bitrate
+      const rate = document.getElementById('rate').value;
+
+      // Collect the keyframe gap
+      const keygap = document.getElementById('keygap').value;
+
+      // Create a MediaStreamTrackProcessor, which exposes frames from the track
+      // as a ReadableStream of VideoFrames, using non-standard Chrome API.
+      let [track] = mediaStream.getVideoTracks();
+      let ts = track.getSettings();
+      const processor = new MediaStreamTrackProcessor(track);
+      inputStream = processor.readable;
+
+      // Create a MediaStreamTrackGenerator, which exposes a track from a
+      // WritableStream of VideoFrames, using non-standard Chrome API.
+      const generator = new MediaStreamTrackGenerator({kind: 'video'});
+      outputStream = generator.writable;
+      document.getElementById('outputVideo').srcObject = new MediaStream([generator]);
+
+      //Create video Encoder configuration
+      const vConfig = {
+         keyInterval: keygap,
+         resolutionScale: 1,
+         framerateScale: 1.0,
+      };
+   
+      let ssrcArr = new Uint32Array(1);
+      window.crypto.getRandomValues(ssrcArr);
+      const ssrc = ssrcArr[0];
+  
+      const config = {
+        alpha: "discard",
+        latencyMode: latencyPref,
+        bitrateMode: "variable",
+        codec: preferredCodec,
+        width: ts.width/vConfig.resolutionScale,
+        height: ts.height/vConfig.resolutionScale,
+        hardwareAcceleration: hw,
+        bitrate: rate, 
+        framerate: ts.frameRate/vConfig.framerateScale,
+        keyInterval: vConfig.keyInterval,
+        ssrc:  ssrc
+      };
+
+      if (mode != "L1T1") {
+        config.scalabilityMode = mode;
+      }
+
+      switch(preferredCodec){
+        case "H264":
+          config.codec = "avc1.42002A";  // baseline profile, level 4.2
+          config.avc = { format: "annexb" };
+          config.pt = 1;
+          break;
+        case "H265":
+          config.codec = "hvc1.2.4.L123.00"; // Main 10 profile, level 4.1, main Tier
```

Comment:
today most encoders support main instead of main10. maybe use hvc1.1.6.L123.00 instead.

Reply:
Fixed.